README.md

```diff
@@ -62,78 +61,97 @@ require 'grell'
 crawler = Grell::Crawler.new
 crawler.start_crawling('http://www.google.com') do |current_page|
   if current_page.status == 500 && current_page.retries == 0
-    crawler.restart
+    crawler.manager.restart
     :retry
   end
 end
 ```
 
-### Restarting PhantomJS
-If you are doing a long crawling it is possible that phantomJS starts failing.
-To avoid that, you can restart it by calling "restart" on crawler.
-That will kill phantom and will restart it. Grell will keep the status of
-pages already visited and pages discovered and to be visited. And will keep crawling
-with the new phantomJS process instead of the old one.
+### Pages' id
+
+Each page has an unique id, accessed by the property 'id'. Also each page stores the id of the page from which we found this page, accessed by the property 'parent_id'.
+The page object generated by accessing the first URL passed to the start_crawling(the root) has a 'parent_id' equal to 'nil' and an 'id' equal to 0.
+Using this information it is possible to construct a directed graph.
+
+
+### Restart and quit
 
-### Selecting links to follow
+Grell can be restarted. The current list of visited and yet-to-visit pages list are not modified when restarting
+but the browser is destroyed and recreated. All cookies are lost. After restarting, crawling is resumed with a
+new browser.
+`crawler.manager.restart` to do a manual restart, look below to do automatic restarts.
 
-Grell by default will follow all the links it finds going to the site
-your are crawling. It will never follow links linking outside your site.
+Grell current crawler can be quit also, that frees the memory taken in Ruby and destroys the PhantomJS process.
+Just do `crawler.manager.quit`
+
+### Options
+
+Grell is highly configurable. Below is the reference of options that can be passed to `Grell:Crawler` class
```

Comment:
This is not marketing material so I think we can do without the first sentence and go straight to the point, for instance:

> The `Grell:Crawler` class can be passed options to customize its behavior: